pkg/envoy/lds/response.go

```diff
@@ -55,13 +65,23 @@ func NewResponse(catalog catalog.MeshCataloger, proxy *envoy.Proxy, _ *xds_disco
 	inboundListener := newInboundListener()
 	if meshFilterChain, err := getInboundInMeshFilterChain(proxyServiceName, cfg); err != nil {
 		log.Error().Err(err).Msgf("Error making in-mesh filter chain for proxy %s", proxy.GetCommonName())
-	} else if meshFilterChain != nil {
+	} else {
+		if !cfg.IsPermissiveTrafficPolicyMode() {
+			// Apply RBAC policies on the inbound filters based on configured policies
+			rbacFilter, err := lb.buildRBACFilter()
```

Comment:
It seems to me that we could pass the result of `ListAllowedInboundServiceAccounts()` to `buildRBACFilter()` here and could also pass `svcAccount`

I think it would read really nice -- `buildRBACFilter( fromInboundServiceAccounts,  toServiceAccount )`

Reply:
We could build it in many different ways. I prefer to keep the rbac filter logic of identifying allowed clients within the rbac.go file.